Llava: add default chat templates (#31691)
huggingface/transformers: docs/source/en/model_doc/llava.md

- Note the model has not been explicitly trained to process multiple images in the same prompt, although this is technically possible, you may experience inaccurate results.

> - For better results, we recommend users to use processor's `apply_chat_template()` method to format your prompt correctly. Each message in the conversation history for chat templates is a dicts with keys "role" and "content". The "content" should be a list of dicts, for "text" and "image" modalaties, as follows:

Reviewer: ```suggestion
- For better results, we recommend users to use the processor's `apply_chat_template()` method to format your prompt correctly. Each message in the conversation history for chat templates is a dictionary with keys "role" and "content". The "content" should be a list of dictionaries, for "text" and "image" modalities, as follows:
```

- Note the model has not been explicitly trained to process multiple images in the same prompt, although this is technically possible, you may experience inaccurate results.

- For better results, we recommend users to use the processor's `apply_chat_template()` method to format your prompt correctly. For that you need to construct a conversation history, passing in a plain string will not format your prompt. Each message in the conversation history for chat templates is a dictionary with keys "role" and "content". The "content" should be a list of dictionaries, for "text" and "image" modalities, as follows:

```python
from transformers import AutoProcessor

processor = AutoProcessor.from_pretrained("llava-hf/llava-1.5-7b-hf")

> conversation = [
>     {
>         "role": "user",
>         "content": [
>             {"type": "image"},
>             {"type": "text", "text": "What’s shown in this image?"},
>             ],
>     },
>     {
>         "role": "assistant",
>         "content": [{"type": "text", "text": "This image shows a red stop sign."},]
>     },
>     {
> 
>         "role": "user",
>         "content": [
>             {"type": "text", "text": "Decsribe the image in more details."},
>             ],
>     },
> ]

Reviewer 2: nit - same here re formatting 

```suggestion
conversation = [
    {
        "role": "user",
        "content": [
            {"type": "image"},
            {"type": "text", "text": "What’s shown in this image?"},
            ],
    },
    {
        "role": "assistant",
        "content": [{"type": "text", "text": "This image shows a red stop sign."},]
    },
    {

        "role": "user",
        "content": [
            {"type": "text", "text": "Describe the image in more details."},
        ],
    },
]
```

text_prompt = processor.apply_chat_template(conversation, add_generation_prompt=True)

# Note that the template simply formats your prompt, you still have to tokenize it and obtain pixel values for your images
print(text_prompt)

> >>> "USER: <image>\n<What’s shown in this image? ASSISTANT: This image shows a red stop sign.</s>USER: Decsribe the image in more details. ASSISTANT:"

Reviewer 2: ```suggestion
>>> "USER: <image>\n<What’s shown in this image? ASSISTANT: This image shows a red stop sign.</s>USER: Describe the image in more details. ASSISTANT:"
```
